src/mapfile.c

```diff
@@ -7296,6 +7298,18 @@ void msApplyDefaultSubstitutions(mapObj *map) {
         layer, &(layer->validation)); /* ...then layer settings... */
     applyLayerDefaultSubstitutions(
         layer, &(map->web.validation)); /* ...and finally web settings */
+
+    // apply any styles referenced in STYLEITEM to the layer
```

Comment:
I'm a bit unconfortable seeing that in msApplyDefaultSubstitutions(). Especially since this function is not called if map->web.validation is set to "immutable". Perhaps having a dedicated function and calling it from towards the end of msLoadMap() would make sense?

Reply:
Moved to its own function `applyStyleItemToLayer` and added calls to `msLoadMapFromString` and `msLoadMap`.